### ISSUE TITLE
Strings deserialized as BigInts

### ISSUE THREAD

**Issue by insulationman** [event-driven-io/Pongo#98]
Strings are interpreted as BigInts if they contain only numbers. This happens when i get data from a collection of projections using findOne.

An error is then thrown when i send these objects in an api response, since BigInts are not properly serialized.

**Comment by oskardudycz** [comment]
@insulationman, indeed, that's an unexpected side effect of the recent changes when I introduced the custom serialiser to support the `_version` property.

I'm going to need to make it smarter, let me outline the challenges:
- `BigInt` support in JSON.parse and stringify is non-existing out of the box, that's why I introduced [JSONSerialiser](https://github.com/event-driven-io/Pongo/blob/1fe83052d1a5001a38d49ad07a2a8b968d3455c7/src/packages/dumbo/src/core/serializer/json/index.ts#L108) (available in `@event-driven-io/dumbo`) that's capable of serialising it,
- when serialising, that's pretty easy as I know the type. When deserialising back it's hard, as the string could be either bigint or just a string (as you mentioned).

There are multiple strategies how to deal with that:
- custom encoding, but that makes filtering more complex and also passing the serialised string as it is,
- adding the custom metadata about the type or using document schema.

I'll probably go with the second option, as it'll be useful also for other cases. But this will take a bit time.

So, as a quick fix, you could use the JSONSerialiser to serialise the payload; if that's urgent, I could also try to provide the fix that changes the default serialisation behaviour, not to (de)serialise version as a string.

@insulationman thoughts?

Also, as I have you here, could you share a bit more about your setup? I'd be happy to get more information on another possible alignment to use Pongo for you. Thank you in advance!

**Comment by insulationman** [comment]
Sounds good! For now i will just validate for number only strings, as the application is simple.

I'm building an mvp for niche service that relates to file uploads and processing. I like the event sourcing methodology so i started using Emmet with Pongo. Fits like a glove.

This is my first serious express/ts backend so I'm learning a lot.
